mlkem/native/x86_64/src/basemul.c

```diff
@@ -8,15 +8,55 @@
 #if defined(MLKEM_NATIVE_ARITH_BACKEND_X86_64_DEFAULT)
 
 #include "consts.h"
+
 #include "poly.h"
 #include "polyvec.h"
 
 #include "arith_native_x86_64.h"
 
-static void poly_basemul_montgomery_avx2(poly *r, const poly *a, const poly *b)
+static const int16_t zetas_avx2[64] = {
+#include "x86_64_zetas_mulcache.i"
+};
+
+#define QINV (-3327) /* q^-1 mod 2^16 */
+
+void poly_mulcache_compute_avx2(poly_mulcache *x, const poly *y)
+{
+  __m256i q, qinv, a0, a1, z, t0, t1, s0, s1, r0, r1;
+
+  q = _mm256_set1_epi16(MLKEM_Q);
+  qinv = _mm256_set1_epi16(QINV);
+
+  for (int j = 0; j < 4; j++)
```

Comment:
This fails the C90 build -- can you move the declaration of `int j` to the top of the function?

Reply:
fixed